### ISSUE TITLE
Broken Package - linux-image-5.15.5 - Unmet dependencies [Pop!_OS 21.04 x86_64]

### ISSUE THREAD

**Issue by BobWazinski** [pop-os/pop#2028]
This issue might be related to the new kernel update but not sure where it is supposed to be placed so I post it here.

Can't perform operating system updates because of a broken package:

> The following packages have unmet dependencies:
 linux-image-5.15.5-76051505-generic : Depends: linux-modules-5.15.5-76051505-generic but it is not installed
E: Unmet dependencies. Try 'apt --fix-broken install' with no packages (or specify a solution).

Tried to update both through sudo apt upgrade and sudo apt full-upgrade, it gives the same error.
Tried to run sudo apt --fix-broken install:

> (Reading database ... 226395 files and directories currently installed.)
Preparing to unpack .../linux-modules-5.15.5-76051505-generic_5.15.5-76051505.202111250933~1638201579~21.04~09f1aa7_amd64.deb ...
Unpacking linux-modules-5.15.5-76051505-generic (5.15.5-76051505.202111250933~1638201579~21.04~09f1aa7) ...
dpkg: error processing archive /var/cache/apt/archives/linux-modules-5.15.5-76051505-generic_5.15.5-76051505.202111250933~1638201579~2
1.04~09f1aa7_amd64.deb (--unpack):
 unable to open '/lib/modules/5.15.5-76051505-generic/kernel/drivers/media/cec/platform/seco/seco-cec.ko.dpkg-new': Operation not perm
itted
Errors were encountered while processing:
 /var/cache/apt/archives/linux-modules-5.15.5-76051505-generic_5.15.5-76051505.202111250933~1638201579~21.04~09f1aa7_amd64.deb
E: Sub-process /usr/bin/dpkg returned an error code (1)

Tried to sudo apt install -f but it resulted in:

> Unpacking linux-modules-5.15.5-76051505-generic (5.15.5-76051505.
202111250933~1638201579~21.04~09f1aa7) ...
dpkg: error processing archive /var/cache/apt/archives/linux-modu
les-5.15.5-76051505-generic_5.15.5-76051505.202111250933~16382015
79~21.04~09f1aa7_amd64.deb (--unpack):
 unable to open '/lib/modules/5.15.5-76051505-generic/kernel/driv
ers/media/dvb-frontends/dvb-pll.ko.dpkg-new': Operation not permi
tted
Errors were encountered while processing:
 /var/cache/apt/archives/linux-modules-5.15.5-76051505-generic_5.
15.5-76051505.202111250933~1638201579~21.04~09f1aa7_amd64.deb
E: Sub-process /usr/bin/dpkg returned an error code (1)

Tried to solve the issue through Synaptic Package Manager as usual but after attempting to reinstall the broken package it gave:

> E: Internal Error, No file name for linux-image-5.15.5-76051505-generic:amd64

Tried to solve the issue using dpkg it gave the original error above.
Tried to install the missing package (linux-modules-5.15.5) but it returned the same error as the force install command.

So honestly confused and have no idea what else to do.

**Comment by leviport** [comment]
What's the output of `apt policy linux-modules-5.15.5-76051505-generic`? I'm seeing it in the repo, so it should be installable. 

> Tried to install the missing package (linux-modules-5.15.5) but it returned the same error as the force install command.

Note that `sudo apt -f install` is the same as `sudo apt --fix-broken install`, not a force install. The package name also has to be exactly correct, so make sure you're trying to install `linux-modules-5.15.5-76051505-generic` and not `linux-modules-5.15.5` (just in case that wasn't abbreviated).

**Comment by BobWazinski** [comment]
> What's the output of `apt policy linux-modules-5.15.5-76051505-generic`? I'm seeing it in the repo, so it should be installable.
> 
> > Tried to install the missing package (linux-modules-5.15.5) but it returned the same error as the force install command.
> 
> Note that `sudo apt -f install` is the same as `sudo apt --fix-broken install`, not a force install. The package name also has to be exactly correct, so make sure you're trying to install `linux-modules-5.15.5-76051505-generic` and not `linux-modules-5.15.5` (just in case that wasn't abbreviated).

The output of apt policy linux-modules-5.15.5-76051505-generic is as follows:

> linux-modules-5.15.5-76051505-generic:
  Installed: (none)
  Candidate: 5.15.5-76051505.202111250933~1638201579~21.04~09f1aa7
  Version table:
     5.15.5-76051505.202111250933~1638201579~21.04~09f1aa7 1001
       1001 http://ppa.launchpad.net/system76/pop/ubuntu hirsute/main amd64 Packages

Regarding the package name, absolutely. Tried the full name of the package. Here's the entire output:

> Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
The following packages were automatically installed and are no longer required:
  calibre-bin fonts-opensymbol gstreamer1.0-gtk3 libabw-0.1-1 libappmenu-gtk3-parser0 libcdr-0.1-1 libchm1 libclucene-contribs1v5
  libclucene-core1v5 libcmis-0.5-5v5 libcolamd2 libe-book-0.1-1 libeot0 libepubgen-0.1-1 libetonyek-0.1-1 libevent-2.1-7
  libexttextcat-2.0-0 libexttextcat-data libfreehand-0.1-1 libjpeg-turbo-progs liblangtag-common liblangtag1 libmhash2
  libmspub-0.1-1 libmwaw-0.3-3 libmythes-1.2-0 libneon27-gnutls libodfgen-0.1-1 liborcus-0.16-0 liborcus-parser-0.16-0
  libpagemaker-0.0-0 libpodofo0.9.7 libqt5designer5 libqt5help5 libqt5positioning5 libqt5quickwidgets5 libqt5test5 libqt5webchannel5
  libqt5webengine-data libqt5webengine5 libqt5webenginecore5 libqt5webenginewidgets5 libraptor2-0 librasqal3 librdf0 libre2-9
  librevenge-0.0-0 libsuitesparseconfig5 libunity-gtk2-parser0 libuno-cppu3 libuno-cppuhelpergcc3-3 libuno-purpenvhelpergcc3-3
  libuno-sal3 libuno-salhelpergcc3-3 libvisio-0.1-1 libwpd-0.10-10 libwpg-0.3-3 libwps-0.4-4 libxmlsec1 libxmlsec1-nss libyajl2
  linux-image-5.13.0-7614-generic linux-modules-5.13.0-7614-generic linux-modules-extra-5.13.0-7614-generic lp-solve optipng
  python3-apsw python3-chm python3-css-parser python3-cssselect python3-cssutils python3-dateutil python3-feedparser
  python3-html2text python3-html5-parser python3-ifaddr python3-mechanize python3-msgpack python3-py7zr python3-pyqt5
  python3-pyqt5.qtsvg python3-pyqt5.qtwebchannel python3-pyqt5.qtwebengine python3-pyqt5.sip python3-regex python3-repoze.lru
  python3-routes python3-texttable python3-webob python3-zeroconf uno-libs-private ure
Use 'sudo apt autoremove' to remove them.
The following NEW packages will be installed:
  linux-modules-5.15.5-76051505-generic
0 upgraded, 1 newly installed, 0 to remove and 10 not upgraded.
1 not fully installed or removed.
Need to get 0 B/69.1 MB of archives.
After this operation, 397 MB of additional disk space will be used.
(Reading database ... 226395 files and directories currently installed.)
Preparing to unpack .../linux-modules-5.15.5-76051505-generic_5.15.5-76051505.202111250933~1638201579~21.04~09f1aa7_amd64.deb ...
Unpacking linux-modules-5.15.5-76051505-generic (5.15.5-76051505.202111250933~1638201579~21.04~09f1aa7) ...
dpkg: error processing archive /var/cache/apt/archives/linux-modules-5.15.5-76051505-generic_5.15.5-76051505.202111250933~1638201579~2
1.04~09f1aa7_amd64.deb (--unpack):
 unable to open '/lib/modules/5.15.5-76051505-generic/kernel/drivers/media/dvb-frontends/ascot2e.ko.dpkg-new': Operation not permitted
Errors were encountered while processing:
 /var/cache/apt/archives/linux-modules-5.15.5-76051505-generic_5.15.5-76051505.202111250933~1638201579~21.04~09f1aa7_amd64.deb
E: Sub-process /usr/bin/dpkg returned an error code (1)

By the way, the Pop!_Shop gives:

> Error while installing package: unable to open '/lib/modules/5.15.5-76051505-generic/kernel/drivers/isdn/hardware/mISDN/speedfax.ko.dpkg-new'

Sorry for the clarity of my stated issue, it is the first time I use GitHub. Kinda the last resort for me. Usually, I manage to fix any arising issues by myself. Not the first time I have had a broken package case. This one is a tough nut though. I hope I won't be 'going Linus' here.

**Comment by leviport** [comment]
Sounds like this article may be helpful: https://support.system76.com/articles/package-manager-pop

The first six commands listed usually fix most package manager issues.

**Comment by BobWazinski** [comment]
> Sounds like this article may be helpful: https://support.system76.com/articles/package-manager-pop
> 
> The first six commands listed usually fix most package manager issues.

Issue SOLVED! Just rebooted my pc and now it runs Kernel 5.15.5! And guess what - All packages are up to date!

Tried to update individual packages which I could through Synaptic Package Manager. Then just like you said tried the first six commands, still couldn't update through the terminal, however, attempting to update through Synaptic Package Manager resolved that problem. 

There were many things I tried in between but the steps above probably were most important.

Thank you for the swift response! You guys are badass!
